homepage/views.py

```diff
@@ -10,11 +10,21 @@
 from django.urls import reverse
 from django.utils.safestring import mark_safe
 from django.utils.crypto import get_random_string
+from bs4 import BeautifulSoup
 
 
 def post(request, id):
     p = get_object_or_404(Post, id=id)
-    return render(request, 'homepage/post.html', {'post': p})
+    text=''
+    q = p.post_content
+    soup = BeautifulSoup(q)
+    s = soup.find_all('h1') + soup.find_all('h2') + soup.find_all('h3') + soup.find_all('h4') + soup.find_all(
```

Comment:
find_all can find multiple tags. In the future find_all may not return lists and so + could fail.